### PR TITLE
Basic healthcheck

### PR DESCRIPTION
**Pull request by KanishkT123**
Issue Number:  User Story 110

## Summary
Adding a healthcheck endpoint and modifying the CI CD pipeline to hit the healthcheck endpoint to check liveness of container.

## Changes

- Added a healthcheck endpoint at `/health`
- Modified CICD to hit `/health`

## Checklist

- [x] Local Tests Passing?
- [x] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [x] This is not a breaking change. If it is, please describe it below.

## Is this a breaking change?
No

## Anything else?
cc: @dnastrain

### CODE REVIEW

File: .github/workflows/build-webapi.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
 
 env:
   DOCKERFILE_PATH: "CarbonAware.WebApi/Dockerfile"
-  HEALTH_ENDPOINT: "0.0.0.0:8080/emissions/bylocation?location=eastus"
```
Comment (juzuluag) [inline .github/workflows/build-webapi.yaml:14]: suggestion: Keep this one too, so we know at least the endpoint works. Maybe creating a new step in the CI that runs the 'data' endpoint would be good to have.

**Review by juzuluag — COMMENTED** [review]
+1. One comment to address.

File: .github/workflows/build-webapi.yaml [hunk 2]
```diff
@@ -11,8 +11,8 @@ on:
...
-  #Health Check Endpoint to be added as another story and replaced here
+  HEALTH_ENDPOINT: "0.0.0.0:8080/health"
+  DATA_ENDPOINT: "0.0.0.0:8080/emissions/bylocation?location=eastus"
```
Comment (bderusha) [inline .github/workflows/build-webapi.yaml:15]: Can you explain what you're hoping to test with this?  I was expecting the health check to replace this endpoint

**Review by bderusha — APPROVED** [review]
Looks good.  Not sure the DATA_ENDPOINT is relevant to the goals of this CI job.  Happy to discuss if you want, but either way it's not a dealbreaker for me.

**Review by juzuluag — APPROVED** [review]
🚢

**Pull request merged**